src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.cs

```diff
@@ -5,27 +5,41 @@
 using System;
 using System.Diagnostics;
 using System.Runtime;
+using System.Runtime.InteropServices;
 using System.Threading;
 using Microsoft.Win32.SafeHandles;
 
 namespace Internal.Runtime.Augments
 {
     public sealed partial class RuntimeThread
     {
+        // Extra bits used in _threadState
+        private const ThreadState ThreadPoolThread = (ThreadState)0x1000;
+        private const ThreadState PublicThreadStateMask = (ThreadState)0x1FF;
+
         [ThreadStatic]
         private static RuntimeThread t_currentThread;
 
+        private int _threadState;
+        private ThreadPriority _priority;
+        private ManagedThreadId _managedThreadId;
+        private string _name;
+        private Lock _lock;
```

Comment:
Could please add a comment describing what this lock is used for and what state it protects?